lldb/tools/lldb-server/Plugins/AMDGPU/AddressSpaces.h

```diff
@@ -0,0 +1,61 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLDB_TOOLS_LLDB_SERVER_PLUGINS_AMDGPU_ADDRESSSPACES_H
+#define LLDB_TOOLS_LLDB_SERVER_PLUGINS_AMDGPU_ADDRESSSPACES_H
+
+#include <cstdint>
+
+namespace lldb_private::lldb_server {
+
+// The address spaces defined here are based on the AMDGPU guide:
+// https://llvm.org/docs/AMDGPUUsage.html#address-space-identifier
+//
+// These are the "dwarf" address spaces, which are used in DWARF operations as
+// opposed to the "llvm" address spaces, which are used in LLVM IR. The amddbg
+// api has its own internal representation of address spaces. We can translate
+// between the two using the following amddbg api call:
+// `amd_dbgapi_dwarf_address_space_to_address_space`.
+//
+enum AddressSpace : uint64_t {
+  /// Default target architecture address space that is used in DWARF operations
+  /// that do not specify an address space. It therefore has to map to the
+  /// global address space so that the DW_OP_addr* and related operations can
+  /// refer to addresses in the program code.
+  DW_ASPACE_LLVM_none = 0,
```

Comment:
We don't need a "no address space" enum. If reading memory through the standard GDB remote memory read packet, then we don't need to define a "no address space" enum value.